### ISSUE TITLE
Using CocoEval - Not getting same results 

### ISSUE THREAD

**Issue by Dicko87** [facebookresearch/detr#319]
Hi there,
I have been using DETR on my own dataset and it works very well. I get a good mAP and Recall on the validation set. My question is, how to I run cocoEval to give me the same or similar results to what it got during model training. For example the model achieved and mAP of 0.89 on the validation set. I then decided to see if I could produce the same results again. I ran the model in eval mode on the dataset and set a confidence threshold > 0.8 and saved the results in a json file. I then used cocoeval and gave the validation set json and my new resFile as inputs and the evaluation results gave me an mAP of 0.6, which isn’t right. How do I go about getting the same or similar results as to what the model achieved originally and consequently how do I adjust the confidence threshold and get the precision-recall curves for these different thresholds. Thank you.

**Comment by alcinos** [comment]
Hello @Dicko87 

The coco evaluation that is performed in the model uses ALL the boxes, without thresholding. That includes boxes for which the class with the highest confidence is the "no-object" class, and in that case we supply the confidence of the second highest scoring class.
This is necessary in order to obtain good AP because it is a ranking metric, hence even object with very low confidence may end up paying true positives and thus boosting the AP score.

However, for many downstream tasks, practitioners will set a threshold as you did in order to have a usable set of boxes. The point here is that AP is not really adequate to compute the performance in that case.

I believe I have answered your question, and as such I'm closing this. Feel free to reach out if you have further concerns.

**Comment by Dicko87** [comment]
Thank you very much for your reply. I should have said that I did try the cocoeval with all predictions (no filter on the confidences) and my mAP result was still a lot lower than what the model showed me. I guess my real question is, what code / steps should I take so that I can get the same results as the why the model gave on the validation set? How to I replicate these figures ? What do I need to do to achieve this? 
So far I ran the model in eval mode, ran predictions on my dataset and saved them into a json file. I then ran the code as shown in the attached image, but the results I got were no where near the same as what the model gave. 
thank you
![AB752642-7B08-4079-B5EF-BEBD19050301](https://user-images.githubusercontent.com/37139958/105013126-d1d40900-5a36-11eb-9eee-83e8694b3a93.jpeg)
